app/models/audit_trail.rb

```diff
@@ -5,63 +5,24 @@ class AuditTrail
   include Mongoid::Userstamp
   mongoid_userstamp user_model: 'User'
 
-  IRRELEVANT_FIELDS = %w[user_ids updated_by_id pledge_sent_by_id last_edited_by_id].freeze
+  IRRELEVANT_FIELDS = %w[user_ids updated_by_id pledge_sent_by_id last_edited_by_id identifier].freeze
+  DATE_FIELDS = %w[appointment_date initial_call_date pledge_generated_at pledge_sent_at fund_pledged_at].freeze
 
   # convenience methods for clean view display
   def date_of_change
     created_at.display_date
   end
 
-  def changed_fields
-    relevant_fields = modified.map do |key, _value|
-      key unless IRRELEVANT_FIELDS.include? key
-    end
-    relevant_fields.compact.map(&:humanize)
-  end
-
-  # TODO: properly render null values like in special circumstances
-  # Something like this should work:
-  # "HAHA I WIN" if (fields.all?{|f| f.blank? || (f.flatten.all? &:blank? if f.respond_to?('each'))})
-
-  def format_dates(hash)
-    for key in ['appointment_date', 'initial_call_date', 'pledge_generated_at']
-      if hash.has_key? key
-        hash[key] = hash[key].display_date
-      end
-    end
-    return hash
-  end
-
-  def format_special_circumstances(hash)
-    hash.each do |key, value|
-      if value.kind_of?(Array)
-        hash[key].reject! { |item| item.blank? }
-        if value.empty?
-          hash[key] = 'not selected'
-        end
-      end
-    end
-    return hash
-  end
-
-  def parse_clinics(hash)
-    if hash.key?('clinic_id')
-      @clinic = Clinic.where(:id => hash["clinic_id"]).first
-      hash["clinic_id"] = @clinic&.name # TODO not having a try here fails the update patient info test for some reason!
-    end
-    return hash
+  def has_changed_fields?
+    modified.reject { |x| IRRELEVANT_FIELDS.include? x }.present?
   end
 
   def changed_from
```

Comment:
I think `changed_from` and `changed_to` are dead code now. Rip them out?

Reply:
great catch, thank you!